zk/lock.go

```diff
@@ -146,5 +162,46 @@ func (l *Lock) Unlock() error {
 	}
 	l.lockPath = ""
 	l.seq = 0
+	// Remove the entry of path of newly created sequential ephemeral znode
+	// this was stored against the parent znode path
+	delete(mapEphermeralSequenceLockPath,l.path)
 	return nil
 }
+
+//Check whether lock got created and response was lost because of network partition failure.
+//It queries zookeeper and scans existing sequential ephemeral znodes under the parent path
+//It finds out that previously requested sequence number corresponds to child having lowest sequence number
+func lockExists(c *Conn, rootPath string, znodePath string) bool {
+	seq, err := parseSeq(znodePath)
+	if err != nil {
+		return false
+	}
+
+	//Scan the existing znodes if there are any
+	children, _, err := c.Children(rootPath)
+	if err != nil {
+		return false
+	}
+
+	lowestSeq := seq
+	prevSeq := -1
+	for _, p := range children {
+		s, err := parseSeq(p)
+		if err != nil {
+			return false
+		}
+		if s < lowestSeq {
+			lowestSeq = s
+		}
+		if s < seq && s > prevSeq {
+			prevSeq = s
+		}
+	}
+
+	if seq == lowestSeq {
+		// Acquired the lock
+		return true
+	} else {
```

Comment:
else is redundant as it's the end of a func

Reply:
Done. Please check